source/FreeRTOS_ND.c

```diff
@@ -1009,6 +1009,10 @@
                        }
 
                        pxICMPHeader_IPv6->ucTypeOfMessage = ipICMP_PING_REPLY_IPv6;
+
+                       /* MISRA Ref 4.14.1 [The validity of values received from external sources]. */
+                       /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#directive-414. */
+                       /* coverity[misra_c_2012_directive_4_14_violation] */
```

Comment:
How about this check - 
if( uxICMPSize > pxNetworkBuffer->xDataLength - ipSIZE_OF_ETH_HEADER - ipSIZE_OF_IPv6_HEADER ) instead of uxNeededSize check. I am trying to see if exception can be avoided.

Reply:
Just replacing `uxNeededSize`, in `if( uxNeededSize > pxNetworkBuffer->xDataLength )`,  is replaced with `( size_t ) ( ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + uxICMPSize );` the violation is not reported. Hence I reported MISRA Ref 4.14.1 as a false positive in the MISRA readme file.